File: serverW2/src/operation/OperationImpl.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package operation;

import java.rmi.RemoteException;
import java.rmi.server.UnicastRemoteObject;
import java.util.ArrayList;
import java.util.List;
import types.*;

/**
 *
 * @author TanatosX
 */

public class OperationImpl extends UnicastRemoteObject implements Operation {
    public OperationImpl ()throws RemoteException{
    }
    static List<Autoparts> lstA = new ArrayList<Autoparts>();
    static List<Brands> lstB = new ArrayList<Brands>();
    static List<Promotions> lstP = new ArrayList<Promotions>();
    static List<Suppliers> lstS = new ArrayList<Suppliers>();
    static List<Warranty> lstW = new ArrayList<Warranty>();
    static {
        lstB.add(new Brands("BMW","Немецкий производитель автомобилей, мотоциклов, двигателей, а также велосипедов"));
        lstB.add(new Brands("Toyota","Крупнейшая японская автомобилестроительная корпорация"));
        lstP.add(new Promotions("+1","06.10.2020",30,"Дополнительная гарантия на год"));
        lstP.add(new Promotions("Toyota30","06.10.2020",60,"Скидка в 30% на запчасти от Toyota"));
        lstS.add(new Suppliers("Аврора","г. Санкт-Петербург, ул. Софийская, д. 2, лит. Б",88123269217L,7804371992L));
        lstS.add(new Suppliers("Орум","г. Санкт-Петербург, ул. Лужская, д. 3",78122488550L,7810155397L));
        lstW.add(new Warranty("Стандартная BMW",12));
        lstW.add(new Warranty("Стандартная Toyota",18));
    }
    

    @Override
    public List<Autoparts> getListA ()throws RemoteException{
        return lstA;
    }
    
    @Override
    public List<Brands> getListB ()throws RemoteException{
        return lstB;  
    }
    
    @Override
    public List<Promotions> getListP ()throws RemoteException{
        return lstP;
    }
    
    @Override
    public List<Suppliers> getListS ()throws RemoteException{
        return lstS;
    }
    
    @Override
    public List<Warranty> getListW ()throws RemoteException{
        return lstW;
    }
    
    @Override
    public List<Autoparts> remListA (int sr)throws RemoteException{
        lstA.remove(sr);
        return lstA;
    }
    
    @Override
    public List<Brands> remListB (int sr)throws RemoteException{
        lstB.remove(sr);
        return lstB;
    }
    
    @Override
    public List<Promotions> remListP (int sr)throws RemoteException{
        lstP.remove(sr);
        return lstP;
    }
    
    @Override
    public List<Suppliers> remListS (int sr)throws RemoteException{
        lstS.remove(sr);
        return lstS;
    }
    
    @Override
    public List<Warranty> remListW (int sr)throws RemoteException{
        lstW.remove(sr);
        return lstW;
    }
    
    @Override
    public List<Autoparts> addNewAutopart (Autoparts item)throws RemoteException{
        lstA.add(item);
        return lstA;
    }
    
    @Override
    public List<Brands> addNewBrand (Brands item)throws RemoteException{
        lstB.add(item);
        return lstB;
    }
    
    @Override
    public List<Promotions> addNewPromotion (Promotions item)throws RemoteException{
        lstP.add(item);
        return lstP;
    }
    
    @Override
    public List<Suppliers> addNewSupplier (Suppliers item)throws RemoteException{
        lstS.add(item);
        return lstS;
    }
    
    @Override
    public List<Warranty> addNewWarranty (Warranty item)throws RemoteException{
        lstW.add(item);
        return lstW;
    }
    
    @Override
    public int getSum ()throws RemoteException{
        int sum = 0;
        for (Autoparts autoparts: lstA)
        {
            sum += autoparts.getKol()*autoparts.getPrice();
        }
        return sum;
    }
}

File: serverW3/serverW3/OperationImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace serverW3
{
    public class OperationImpl : MarshalByRefObject, Operation
    {
        public static List<Autoparts> lstA = new List<Autoparts>();
        public static List<Brands> lstB = new List<Brands>();
        public static List<Promotions> lstP = new List<Promotions>();
        public static List<Suppliers> lstS = new List<Suppliers>();
        public static List<Warranty> lstW = new List<Warranty>();

        public List<Autoparts> getListA (){
        return lstA;
        }
    
        public List<Brands> getListB (){
        return lstB;  
        }
    
        public List<Promotions> getListP (){
        return lstP;
        }
    
        public List<Suppliers> getListS (){
        return lstS;
        }
    
        public List<Warranty> getListW (){
        return lstW;
        }
    
        public List<Autoparts> remListA (int sr){
            lstA.RemoveAt(sr);
            return lstA;
        }
    
        public List<Brands> remListB (int sr){
            lstB.RemoveAt(sr);
            return lstB;
        }
    
        public List<Promotions> remListP (int sr){
            lstP.RemoveAt(sr);
            return lstP;
        }
    
        public List<Suppliers> remListS (int sr){
            lstS.RemoveAt(sr);
            return lstS;
        }
    
        public List<Warranty> remListW (int sr){
            lstW.RemoveAt(sr);
            return lstW;
        }
    
        public List<Autoparts> addNewAutopart (Autoparts item){
        lstA.Add(item);
        return lstA;
        }
    
        public List<Brands> addNewBrand (Brands item){
        lstB.Add(item);
        return lstB;
        }
    
        public List<Promotions> addNewPromotion (Promotions item){
        lstP.Add(item);
        return lstP;
        }
    
        public List<Suppliers> addNewSupplier (Suppliers item){
        lstS.Add(item);
        return lstS;
        }
    
        public List<Warranty> addNewWarranty (Warranty item){
        lstW.Add(item);
        return lstW;
        }
    
        public int getSum (){
            int sum = 0;
            foreach (Autoparts autoparts in lstA)
            {
                sum += autoparts.getKol()*autoparts.getPrice();
            }
            return sum;
        }
    }
}
